src/v/coproc/script_context_router.cc

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "script_context_router.h"
+
+#include "reflection/absl/flat_hash_map.h"
+
+namespace reflection {
+
+ss::future<>
+async_adl<coproc::read_context>::to(iobuf& out, coproc::read_context&& rctx) {
+    reflection::serialize(
+      out, rctx.absolute_start, rctx.last_read, rctx.last_acked);
+    return ss::now();
+}
+
+ss::future<coproc::read_context>
+async_adl<coproc::read_context>::from(iobuf_parser& in) {
+    coproc::read_context rctx;
+    rctx.absolute_start = adl<model::offset>{}.from(in);
+    rctx.last_read = adl<model::offset>{}.from(in);
+    rctx.last_acked = adl<model::offset>{}.from(in);
+    return ss::make_ready_future<coproc::read_context>(std::move(rctx));
```

Comment:
do you need versioning?

Reply:
Possibly in the future, suggestions? Maybe have a version field for the router itself?

Reply:
or rather the `coproc::source` struct?

Reply:
what are the scenarios you care about going forward with respect to on-wire and on-disk data? in principle it shouldn't be different from other cases where you have used versioning in combination with serialization.

Reply:
I don't really foresee this changing unless maybe some new feature necessitates a change.

Reply:
it looks like you added a version in the latest force push, but didn't use it to in any verification checks. how is the version that you added related to the claim here that there won't ever be any changes?